[Transform] Implement per-transform num_failure_retries setting. (#87361)
elastic/elasticsearch: docs/reference/rest-api/common-parms.asciidoc

The default value is `500`.
end::transform-settings-max-page-search-size[]

tag::transform-settings-num-failure-retries[]

> Defines the number of retries on a recoverable failure before giving up.

Reviewer: Small adjustment, so the description is more in line with `xpack.transform.num_transform_failure_retries`.
```suggestion
Defines the number of retries on a recoverable failure before the {transform} task is 
marked as `failed`.
```

Author: Done.

The minimum value is `0` and the maximum is `100`.

> Additionally, a special value `-1` can be used to denote infinity, i.e.: the transform will never give up on retrying

Reviewer: ```suggestion
`-1` can be used to denote infinity. In this case, the {transform} never gives up on retrying
```

Author: Done.
